app/src/pages/inside/settingsPage/settingTabs/demoDataTab/demoDataTabForm/demoDataTabForm.jsx

```diff
@@ -0,0 +1,135 @@
+import styles from './demoDataTabForm.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  postfixInputPlaceholder: {
+    id: 'DemoDataTabForm.postfixInputPlaceholder',
+    defaultMessage: 'Enter postfix',
+  },
+  generateButtonTitle: {
+    id: 'DemoDataTabForm.generateButtonTitle',
+    defaultMessage: 'Generate demo data',
+  },
+  preloaderInfo: {
+    id: 'DemoDataTabForm.preloaderInfo',
+    defaultMessage:
+      'Data generation is started. The process can take several minutes, please wait.',
+  },
+  generateDemoDataSuccess: {
+    id: 'SuccessMessages.generateDemoDataSuccess',
+    defaultMessage: 'Demo data has been generated',
+  },
+});
+
+@reduxForm({
+  form: 'demoDataTabForm',
+  validate: ({ demoDataPostfix }) => ({
+    demoDataPostfix:
+      (!demoDataPostfix || !validate.demoDataPostfix(demoDataPostfix)) && 'demoDataPostfixHint',
+  }),
+})
+@connect(
+  (state) => ({
+    projectId: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+@injectIntl
+export class DemoDataTabForm extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    projectId: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    reset: PropTypes.func.isRequired,
+  };
+
+  state = {
+    isLoading: false,
+  };
+
+  onFormSubmit = (formData) => {
+    const { intl, projectId, reset } = this.props;
+    const data = {
+      isCreateDashboard: 'true',
+      postfix: formData.demoDataPostfix,
+    };
+
+    this.setState({
+      isLoading: true,
+    });
+    fetch(URLS.generateDemoData(projectId), { method: 'POST', data })
+      .then(() => {
+        this.props.showNotification({
+          message: intl.formatMessage(messages.generateDemoDataSuccess),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.setState({
+          isLoading: false,
+        });
+        reset('demoDataTabForm');
+      })
+      .catch((e) => {
+        const error = (e.response && e.response.data && e.response.data.message) || e.message;
+        this.props.showNotification({
+          messageId: 'failureDefault',
+          type: NOTIFICATION_TYPES.ERROR,
+          values: { error },
+        });
+        this.setState({
+          isLoading: false,
+        });
+      });
+  };
+
+  render() {
+    const { intl, handleSubmit } = this.props;
+
+    return (
+      <form onSubmit={handleSubmit(this.onFormSubmit)}>
+        <div className={cx('input-wrapper')}>
+          <FieldProvider name="demoDataPostfix" disabled={this.state.isLoading}>
+            <FieldErrorHint>
+              <Input placeholder={intl.formatMessage(messages.postfixInputPlaceholder)} />
+            </FieldErrorHint>
+          </FieldProvider>
+        </div>
+        <div className={cx('generate-button-container')}>
+          <div className={cx('generate-button')}>
```

Comment:
Why do you create two divs?  Please don't creale a lot of containers if it posible to make murkup without them

Reply:
Fixed